lambdas/gro-convert-to-json/src/main/java/uk/gov/di/data/lep/GroConvertToJson.java

```diff
@@ -107,4 +91,28 @@ private String getAuthorisationToken() {
             throw new AuthException("Failed to send authorisation request", e);
         }
     }
+
+    private String convertXmlDataToJson(String xmlData, String authorisationToken) {
+        try {
+            var deathRegistrationGroup = xmlMapper.readValue(xmlData, DeathRegistrationGroup.class);
+            var records = deathRegistrationGroup.deathRegistrations();
+
+            if (records == null) {
+                if(deathRegistrationGroup.recordCount() == 0) {
```

Comment:
Do we have anywhere that checks the number of records is equal to the record count?

Reply:
This is listed as a separate ticket so will be on a different branch/in a different PR :) (will be ready v soon)